File: cases/taint/java/model-entrypoint-selectivity-positive/Handler.java
package dataflowbench.taint;

final class Handler {
    static void dfb_sink(String value) { }  // DFB-SINK: model-entrypoint-selectivity-sink

    void onDeclared(String input) {  // DFB-SOURCE: model-entrypoint-selectivity-input
        dfb_sink(input);
    }

    void onUndeclared(String input) {
        dfb_sink("clean");
    }
}

File: cases/taint/javascript/model-entrypoint-selectivity-positive/Handler.js
function dfb_sink(value) {} // DFB-SINK: model-entrypoint-selectivity-sink

const Handler = {
  onDeclared: function onDeclared(input) { // DFB-SOURCE: model-entrypoint-selectivity-input
    dfb_sink(input);
  },
  onUndeclared: function onUndeclared(input) {
    dfb_sink("clean");
  }
};
